src/Package/Impl/DataInspect/Viewers/GridViewerBase.cs

```diff
@@ -32,7 +33,12 @@ internal abstract class GridViewerBase : ViewerBase, IObjectDetailsViewer {
             if (evaluation != null) {
                 await VsAppShell.Current.SwitchToMainThreadAsync();
 
-                var id = _toolWindowIdBase + evaluation.GetHashCode() % (Int32.MaxValue - _toolWindowIdBase);
+                var id = Math.Abs(_toolWindowIdBase + expression.GetHashCode() % (Int32.MaxValue - _toolWindowIdBase));
+
+                var existingPane = ToolWindowUtilities.FindWindowPane<VariableGridWindowPane>(id);
+                var frame = existingPane?.Frame as IVsWindowFrame;
+                frame?.CloseFrame(0);
```

Comment:
Perhaps. I need to check if it actually updates the data or if the data is detached.

---
In reply to: [63252280](https://github.com/Microsoft/RTVS/pull/1765#discussion_r63252280) [](ancestors = 63252280)